sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -75,31 +112,46 @@ export interface MessageHandlerOptions {
     maxMessageAutoRenewLockDurationInSeconds?: number;
 }
 
-export { MessagingError }
-
 // @public
-export interface OnError {
-    (error: MessagingError | Error): void;
+export interface MessageHandlers<ContextT> {
+    processError(err: Error): Promise<void>;
+    processMessage(message: ReceivedMessage, context: ContextT): Promise<void>;
 }
 
 // @public
-export interface OnMessage {
-    (message: ServiceBusMessage): Promise<void>;
-}
+export type MessageIterator<ContextT> = AsyncIterable<MessageAndContext<ContextT>>;
+
+export { MessagingError }
 
-// Warning: (ae-forgotten-export) The symbol "Client" needs to be exported by the entry point index.d.ts
-//
 // @public
-export class QueueClient implements Client {
+export interface NonSessionReceiver<LockModeT extends "peekLock" | "receiveAndDelete"> {
     close(): Promise<void>;
-    createReceiver(receiveMode: ReceiveMode): Receiver;
-    createReceiver(receiveMode: ReceiveMode, sessionOptions: SessionReceiverOptions): SessionReceiver;
-    createSender(): Sender;
-    readonly entityPath: string;
-    static getDeadLetterQueuePath(queueName: string): string;
-    readonly id: string;
-    peek(maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-    peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
+    diagnostics: {
+        peek(maxMessageCount?: number): Promise<ReceivedMessage[]>;
+        peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessage[]>;
```

Comment:
In .NET, we are using PeekAt for peeking by sequence number.

Reply:
Leaving this for now - I think we left the arch review with the idea that this is an experiment.